mat64/vector.go

```diff
@@ -137,6 +137,31 @@ func (v *Vector) ScaleVec(alpha float64, a *Vector) {
 	blas64.Scal(n, alpha, v.mat)
 }
 
+// AddScaledVec adds the vectors a and alpha*b, placing the result in the receiver.
+func (v *Vector) AddScaledVec(a *Vector, alpha float64, b *Vector) {
+	ar := a.Len()
+	br := b.Len()
+
+	if ar != br {
+		panic(ErrShape)
+	}
+
+	v.reuseAs(ar)
+
+	switch {
+	case v == a && v == b: // v <- v + alpha * v = (alpha + 1) * v
+		blas64.Scal(ar, alpha+1, v.mat)
+	case v == a && v != b: // v <- v + alpha * b
+		blas64.Axpy(ar, alpha, b.mat, v.mat)
+	case v != a && v == b: // v <- a + alpha * v
+		blas64.Scal(ar, alpha, v.mat)
```

Comment:
I would guess based on the current coding of packages, that the two blas64 calls is notably slower than writing the one blas call. For large vectors, cache misses are everything (go look at our benchmarks for Daxpy vs OpenBLAS, even with all of the assembly tricks blas/native is the same performance). Double implementing everything seems bad as well. 

I would propose that we add the fast-path if both vectors have inc1.
````
case v != a && v == b: // v <- a + alpha * v
if v.mat.Inc == 1 && a.mat.Inc == 1{ 
       for i, f := range a {
              v[i] *= alpha
              v[i] += f
       }
     return
}
blas64.Scal(ar, alpha, v.mat)
blas64.Axpy(ar, 1, a.mat, v.mat)
````
This at least short circuits the most common case.